### PR TITLE
[Bugfix] Fix bug create MV with explain query and add compatibility

### LINKED ISSUE #10347: Bad `CREATE MATERIALIZED VIEW` will cause FE down
### Steps to reproduce the behavior (Required)
```
mysql> create table customer (c_name varchar(100), c_phone int(18)) DUPLICATE KEY(c_name) DISTRIBUTED BY HASH(c_name) BUCKETs 4 PROPERTIES("replication_num" = "1");
Query OK, 0 rows affected (0.02 sec)

mysql> CREATE MATERIALIZED VIEW mv_customer_sumc_phone AS
    -> EXPLAIN SELECT c_name, MAX(c_phone)
    -> FROM customer
    -> GROUP BY c_name;
Query OK, 0 rows affected (0.06 sec)
```

### Expected behavior (Required)
Execute failed with error messages.

### Real behavior (Required)
It seems to execute well, but later all FE are down when replaying the journal.

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.3.0

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] bugfix
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #10347

## Problem Summary(Required) ：

This problem is handled without throwing an exception. Rather, as a hint, when users come to you for an import or query, suggest that they drop the materialized view and recreate to fix the problem.
## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 5 (20.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/MetaUtils.java|0|3|00.00%|[125, 129, 132]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|2|50.00%|[1186]|

File: fe/fe-core/src/main/java/com/starrocks/sql/common/MetaUtils.java [hunk 1]
```diff
@@ -121,9 +122,14 @@ public static Map<String, Expr> parseColumnNameToDefineExpr(OriginStatement orig
             stmt.setIsReplay(true);
             return stmt.parseDefineExprWithoutAnalyze(originStmt.originStmt);
         } catch (Exception e) {
-            throw new IOException("error happens when parsing create materialized view stmt use old parser:" +
+            LOG.warn("error happens when parsing create materialized view stmt use old parser:" +
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/common/MetaUtils.java:125]:
```suggestion
            LOG.warn("error happens when parsing create materialized view stmt:{} use old parser" +
```

**Review by ABingHuang — CHANGES_REQUESTED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 6 (16.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/MetaUtils.java|0|4|00.00%|[113, 126, 130, 133]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|2|50.00%|[1186]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 6 (16.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/MetaUtils.java|0|4|00.00%|[112, 125, 129, 132]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|2|50.00%|[1186]|

**Review by ABingHuang — DISMISSED** [review]
LGTM

**Review by gengjun-git — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 2]
```diff
@@ -1181,6 +1181,11 @@ public ParseNode visitCreateMaterializedViewStatement(
         String comment =
                 context.comment() == null ? null : ((StringLiteral) visit(context.comment().string())).getStringValue();
         QueryStatement queryStatement = (QueryStatement) visit(context.queryStatement());
+
+        if (queryStatement.isExplain()) {
+            throw new IllegalArgumentException("Materialized view does not support explain query");
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:1186]: AstBuilder is to build ast node, Semantic check should add to xxxAnalyzer

**Review by HangyuanLiu — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by Astralidea** [comment]
@Mergifyio backport branch-2.4

**Comment by Astralidea** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**